### ISSUE TITLE
Building from Dockerfile fails if postgres not running 

### ISSUE THREAD

**Issue by TastyPi** [mastodon/mastodon#12288]
### Expected behaviour

Running `docker build .` should successfully build a Docker image.

### Actual behaviour

Build fails with the error in https://gist.github.com/TastyPi/fe66b1fadf797f4c6864ccb5a302db7f.

Error is caused by no postgres server running on the machine.

### Steps to reproduce the problem

Run `docker build .` on a machine with no postgres server running.

I originally came across this bug when attempting to build a container image with Google Cloud Build, but I have also reproduced this with my desktop not running a postgres server. 

### Specifications

Found in commit 685b0db

**Comment by exstral** [comment]
Also came across this problem when building a docker image for the hometown fork. 

Had to remove the "precompile assets" step. Seems that when compiling a template it automatically tries to fetch the `site_title` setting from the database. But there is none yet.

Not sure how to disable this properly. Wondering how they manage to build their official image.

**Comment by exstral** [comment]
Okay removing the "precompile assets" of course made the image unusable, so don't do that 😄 

After a lot of attempts I've found that the actual error I'm having is this:
```
Browserslist: caniuse-lite is outdated. Please run next command `yarn upgrade`
```

This is probably because of some dependency hell in yarn/npm still present in the [hometown fork](https://github.com/hometown-fork/hometown). Will create an issue there instead.

**Comment by TastyPi** [comment]
I ended up fixing it for me by commenting out the `render_static_page 'errors/500'` line in lib/tasks/assets.rake since that's the only precompile step which attempts to access the database. IIRC you also need to make sure the nginx config does not point to the static file which is no longer being generated. I'm also using hometown, that was the only fix I had to do but I haven't tried building for a couple of months.
